File: python/TokoKomponen.py
from TokoKomputer import TokoKomputer

class TokoKomponen(TokoKomputer):
    def __init__(self, nama_komponen="", merk_komponen="", harga_komponen="", **kwargs):
        super().__init__(**kwargs)
        self.nama_komponen = nama_komponen
        self.merk_komponen = merk_komponen
        self.harga_komponen = harga_komponen

    # Getter & Setter
    def set_nama_komponen(self, nama_komponen):
        self.nama_komponen = nama_komponen

    def get_nama_komponen(self):
        return self.nama_komponen

    def set_merk_komponen(self, merk_komponen):
        self.merk_komponen = merk_komponen

    def get_merk_komponen(self):
        return self.merk_komponen

    def set_harga_komponen(self, harga_komponen):
        self.harga_komponen = harga_komponen

    def get_harga_komponen(self):
        return self.harga_komponen

    # Add data komponen
    def add_data_komponen(self, nama_toko, alamat_toko, notel_toko,
                          kode_pc, spesifikasi_pc, harga_pc,
                          nama_komponen, merk_komponen, harga_komponen):
        super().add_data_komputer(nama_toko, alamat_toko, notel_toko, kode_pc, spesifikasi_pc, harga_pc)
        self.nama_komponen = nama_komponen
        self.merk_komponen = merk_komponen
        self.harga_komponen = harga_komponen

File: java/TokoKomponen.java
public class TokoKomponen extends TokoKomputer {
    private String namaKomponen;
    private String merkKomponen;
    private String hargaKomponen;

    public TokoKomponen() {}

    // Setter & Getter
    public void setNamaKomponen(String namaKomponen) {
        this.namaKomponen = namaKomponen;
    }
    public String getNamaKomponen() {
        return namaKomponen;
    }

    public void setMerkKomponen(String merkKomponen) {
        this.merkKomponen = merkKomponen;
    }
    public String getMerkKomponen() {
        return merkKomponen;
    }

    public void setHargaKomponen(String hargaKomponen) {
        this.hargaKomponen = hargaKomponen;
    }
    public String getHargaKomponen() {
        return hargaKomponen;
    }

    // Add data komponen
    public void addDataKomponen(String namaToko, String alamatToko, String notelToko,
                                String kodePC, String spesifikasiPC, String hargaPC,
                                String namaKomponen, String merkKomponen, String hargaKomponen) {
        super.addDataKomputer(namaToko, alamatToko, notelToko, kodePC, spesifikasiPC, hargaPC);
        this.namaKomponen = namaKomponen;
        this.merkKomponen = merkKomponen;
        this.hargaKomponen = hargaKomponen;
    }
}
